### PR TITLE
feat/populate-user-profile

### PR DESCRIPTION
**Pull request by SabitRaiyan**
For business area, I need to still add the functionality to submit the updated details on selecting an option. Also, for the seniority, I have pulled in all the possible enum values for now, but this will be made to only contain the appropriate enum value.

### CODE REVIEW

**Comment by michaellimair** [comment]
# [Codecov](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim) Report
> Merging [#64](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim) (591242b) into [main](https://codecov.io/gh/michaellimair/cs261-group8/commit/449ba609c88bdb70da3b5a395c9164b081079bae?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim) (449ba60) will **not change** coverage.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim)

```diff
@@            Coverage Diff            @@
##              main       #64   +/-   ##
=========================================
  Coverage   100.00%   100.00%           
=========================================
  Files           17        17           
  Lines          370       370           
=========================================
  Hits           370       370           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim). Last update [449ba60...591242b](https://codecov.io/gh/michaellimair/cs261-group8/pull/64?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Michael+Lim).

File: frontend/src/components/user-profile-components/BusinessAreas.tsx [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
-      <option>Group Audit</option>
-      <option>Asset Management</option>
-    </Select>
-  </FormControl>
-);
+const BusinessAreas: FC = () => {
+  const [items, setItems] = useState([]);
+  const { user } = useUser();
+  const [areaID, setAreaID] = useState<number>(0);
+  const [businessAreaList, setBusinessAreaList] = useState<IBusinessArea[]>([]);
```
Comment (michaellimair) [inline frontend/src/components/user-profile-components/BusinessAreas.tsx]: Can we use react-query for this instead?

```
const { businessArea } = useQuery('business-area-list', () => httpClient.businessArea.listBusinessAreas())
```

Also, for business area by ID, we can probably do something like this
```
const areaID = user?.profile?.business_area ?? null;
```
As far as I can remember, the user business area is already in the right format.

File: frontend/src/pages/identity/UserProfile.tsx [hunk 2]
```diff
@@ -13,128 +14,185 @@ import {
...
+  const [flag, setFlag] = useState<string>('');
+  const [businessLabel, setBusinessLabel] = useState('');
+
+  const onSubmitHandler = () => {
+  };
+
+  useEffect(() => {
+    const countryCode = user?.profile.country;
+    if (countryCode) {
+      httpClient.country.getCountryByCode(countryCode)
```
Comment (michaellimair) [inline frontend/src/pages/identity/UserProfile.tsx]: Same here, use `react-query`

Comment (michaellimair) [reply frontend/src/pages/identity/UserProfile.tsx]: Another option would be to create a custom `useCountryByCode` hook

File: frontend/src/components/user-profile-components/BusinessAreas.tsx (hunk 1 above)
Comment (SabitRaiyan) [reply frontend/src/components/user-profile-components/BusinessAreas.tsx]: I have done this now. However, I have one last issue. I am still not sure how to submit the data despite using the common form hook. I need some help on this. I tried to do this but the page did not rerender the correct business area upon change.

**Pull request merged**